src/insertsort.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals
+from random import choice
+
+
+def insertion_sort(my_list):
+    """Simple sorting algorithm that builds a sorted list
+       one item at a time.
+    """
+    for i in range(1, len(my_list)):
+        current = my_list[i]
+        position = i
+        while((position > 0) and (my_list[position - 1] > current)):
+            my_list[position] = my_list[position - 1]
+            position = position - 1
+        if position != i:
+            my_list[position] = current
+
+
+def _random_list(n):
```

Comment:
This function will create a random list with no duplicate values. You can do this much more simply with `random.sample.`
```python
from random import sample

random_list = sample(range(1000), 100)
```
This would give a list with 100 items between 0 and 999.